src/Flow.jl

```diff
@@ -61,21 +61,27 @@ upperBoundary!(r,u,Φ,ν,i,j,N,::Val{true}) = @loop r[I-δ(j,I),i] -= Φ[CIj(j,I
 
 using EllipsisNotation
 """
-    accelerate!(r,t,g)
+    accelerate!(r,dt,g)
 
-This function adds a uniform acceleration field `g` at time `t` to `r`.
-If `g ≠ nothing`, then `g(i,t)=dUᵢ/dt`.
+Add a uniform acceleration `gᵢ+dUᵢ/dt` at time `t=sum(dt)` to field `r`.
 """
-accelerate!(r,t,g::Function,::Tuple) = for i ∈ 1:last(size(r))
+accelerate!(r,dt,g::Function,::Tuple,t=sum(dt)) = for i ∈ 1:last(size(r))
     r[..,i] .+= g(i,t)
 end
-accelerate!(r,t,g::Nothing,U::Function) = for i ∈ 1:last(size(r))
+accelerate!(r,dt,g::Nothing,U::Function,t=sum(dt)) = for i ∈ 1:last(size(r))
     r[..,i] .+= ForwardDiff.derivative(τ->U(i,τ),t)
 end
-accelerate!(r,t,g::Function,U::Function) = for i ∈ 1:last(size(r))
+accelerate!(r,dt,g::Function,U::Function,t=sum(dt)) = for i ∈ 1:last(size(r))
     r[..,i] .+= g(i,t) + ForwardDiff.derivative(τ->U(i,τ),t)
 end
-accelerate!(r,t,::Nothing,::Tuple) = nothing
+accelerate!(r,dt,::Nothing,::Tuple) = nothing
+"""
+    BCTuple(U,dt,N)
+
+Return BC tuple `U(i∈1:N, t=sum(dt))`.
+"""
+BCTuple(f::Function,dt,N,t=sum(dt))=ntuple(i->f(i,t),N)
+BCTuple(f::Tuple,dt,N)=f
```

Comment:
These two definitions fo `BCTuple` are almost the same at lines 168-169 and the code does not compile for me.